spyderlib/widgets/findreplace.py

```diff
@@ -198,6 +198,39 @@ def show(self):
         self.visibility_changed.emit(True)
         if self.editor is not None:
             text = self.editor.get_selected_text()
+            
+            # If no text is highlighted for search, use whatever word is under the cursor
+            if len(text)==0:
+                try:
+                    # Get line text and cursor position
+                    line_number, index = self.editor.get_cursor_line_column()
+                    block = self.editor.document().findBlockByNumber(line_number)
+                    data = str( block.text() )
+                    ibeg = index
+                    # Limit the search for the beginning of the word to a "reasonable" amount
+                    while (ibeg>0) and (ibeg>index-80):

```

Comment:
What is the idea of this "reasonable" amount, can you elaborate on this please?

Also... it should be `while ibeg > 0 and ibeg > (index-80):`